journal: store CreationTime for VolumeGroups

nixpanic opened the pull request:
VolumeGroupSnapshots expect to have a CreationTime attribute. This can not be obtained by any (go-ceph) API calls, so it need to be stored in the journal.

nixpanic committed:
rbd: have GetCreationTime() return a time.Time struct

Do not use protobuf types when there is no need. Just use the standard
time.Time format instead.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
journal: store CreationTime for VolumeGroups

Signed-off-by: Niels de Vos <ndevos@ibm.com>

Rakshith-R commented on internal/rbd/snapshot.go:
```diff
@@ -118,6 +121,16 @@ func (rbdSnap *rbdSnapshot) toVolume() *rbdVolume {
 	}
 }
 
+func (rbdSnap *rbdSnapshot) ToCSI(ctx context.Context) (*csi.Snapshot, error) {
```
Let's make use of it [here](https://github.com/ceph/ceph-csi/blob/devel/internal/rbd/controllerserver.go#L1238) ?

nixpanic replied:
oh, sure, this function actually was not required to be included, it is part of my VolumeGroupSnapshot work. I'll include it here so that it all gets a little cleaner.

Madhu-1 commented on internal/csi-addons/rbd/replication.go:
```diff
@@ -746,8 +746,8 @@ func (rs *ReplicationServer) ResyncVolume(ctx context.Context,
-func timestampToString(st *timestamppb.Timestamp) string {
-	return fmt.Sprintf("seconds:%d nanos:%d", st.GetSeconds(), st.GetNanos())
+func timestampToString(st *time.Time) string {
+	return fmt.Sprintf("seconds:%d nanos:%d", st.Unix(), st.Nanosecond())
```
can we check on this one, Does Unix is equivalent to GetSeconds()?

nixpanic replied:
Yes
- https://pkg.go.dev/google.golang.org/protobuf/types/known/timestamppb#Timestamp
- https://pkg.go.dev/time#Time.Unix

nixpanic committed:
rbd: convert rbdVolume to rbdSnapshot

After cloning the RBD snapshot, an rbdVolume is returned for the
CSI.Snapshot object. In order to use the rbdSnapshot.ToCSI() function,
the rbdVolume needs to be converted (back) to an rbdSnaphot.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic commented:
/test ci/centos/k8s-e2e-external-storage/1.28

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.29

nixpanic commented:
@iPraveenParihar this adds the creation time to the journal, maybe you need to add that to #4750 as well?

nixpanic commented:
@mergifyio rebase

iPraveenParihar reviewed (approved):
Thanks!

nixpanic commented:
@mergifyio queue
